### ISSUE TITLE
"Template Should Not Contain Blanks" tests for blanks but some Azure Resources necessitate a blank

### ISSUE THREAD

**Issue by mgeorgebrown89** [Azure/arm-ttk#22]
I'm not sure if this is somehow actually just my fault, but when using a userAssignedIdentity, you have to declare it in the ARM template like so:

     "identity": {
                "type": "UserAssigned",
                "userAssignedIdentities": {
                    "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', 
    variables('integration').userAssignedIdentity.name)]": {}
                }
            },

That little `{}` at the end there is causing the "Template Should Not Contain Blanks" to fail. This seems odd. 

Perhaps the test should check strictly for newline characters?

**Comment by mgeorgebrown89** [comment]
Oh, and now I see part of the problem. There are exceptions in Template-Should-Not-Contain-Blanks.test.ps1 here:

    $PropertiesThatCanBeEmpty = 'resources', 
                            'outputs', 
                            'variables', 
                            'parameters', 
                            'functions', 
                            'properties', 
                            'defaultValue', # enables optional parameters
                            'accessPolicies',  # keyVault requires this
                            'value', # Microsoft.Resources/deployments - passing empty strings to a nested deployment
                            'promotionCode', # Microsoft.OperationsManagement/soltuions/plan object
                            'inputs' # Microsoft.Portal/dashboard

But in this particular case, the property is dynamic, making it difficult to just add to this list.

**Comment by bmoore-msft** [comment]
Thanks for the bug... - you're right this one is tricky, we should be able to detect if the property is an expression though in this case it could be a literal too (e.g. GUID).  That might unblock, but not sure if it's sustainable.

**Comment by marrobi** [comment]
I've got this same issue, and another on the same rule...

 Can you think of another way to do this? I need to conditionally join three subnets (this is one example, I have another with security rules):

```
"subnets": "[union(variables('defaultSubnets'),if(parameters('deployVnetGateway'), variables('gatewaySubnet'),json('[]')),if(parameters('deployBastion'), variables('bastionSubnet'),json('[]')))]",
```

**Comment by bmoore-msft** [comment]
@marrobi - can you file a separate issue for your scenario?  As a workaround can you change json('[]') to json('null')?  That might trigger different behavior if the vnet already has subnets, not 100% certain.

**Comment by marrobi** [comment]
I'm hitting the `userAssignedIdentities` issue on a number of templates. Possible to add an exception for `userAssignedIdentities`? Thanks.

**Comment by bmoore-msft** [comment]
Fixed in #65
